app.ts

```diff
@@ -0,0 +1,252 @@
+import { ajax, css } from "jquery";
+
+class RecordManager {
+	firstNumber: number;
+	lastNumber: number;
+	recordCount: number;
+
+	constructor() {
+		this.firstNumber = 0;
+		this.lastNumber = 0;
+		this.recordCount = 0;
+		this.initialize();
+	}
+
+	initialize() {
+		this.createTableHeader();
+		dataManager.fetchRecordCount()
+			.then(count => {
+				this.recordCount = count - 1;
+				this.updateAndDisplayRecords();
+				this.recordEventHandlers();
+				this.handleResize();
+			})
+			.catch(err => {
+				alert('Error fetching and displaying the table records, reload the page');
+				throw err;
+			});
+	}
+
+	/** Initializes the table head */
+	createTableHeader() {
+		return dataManager.fetchColumns()
+			.then(columns => {
+				for (const col of columns) {
+					$(".head").append(`<th>${col}</th>`);
+				}
+			})
+			.catch(err => {
+				alert('Error creating table heading');
+				throw err;
+			});
+	}
+
+	/** calculates the number of rows that can fit the screen */
+	calculatingRows(): number {
```

Comment:
```suggestion
	getNumberOfRows(): number {
```
With a function name like that it is more descriptive of what it does. calculatingRows sounds like a boolean check like isCalculatingRows? Rather give it the verb of what it is going to do. I mean, since this function returns something how about get? How it got it is not maybe important. It calculated it, someone else calculated it? We just want the number of rows.